src/Image.php

```diff
@@ -41,20 +41,42 @@
 
 if (!extension_loaded("vips")) {
     if (!dl('vips.' . PHP_SHLIB_SUFFIX)) {
-        echo "vips: unable to load vips extension\n"; 
+        echo "vips: unable to load vips extension\n";
     }
 }
 
 /**
- * Image represents an image. 
+ * Image represents an image.
  *
  * @category  Images
- * @package   Vips
+ * @package   Jcupitt\Vips
  * @author    John Cupitt <jcupitt@gmail.com>
  * @copyright 2016 John Cupitt
  * @license   https://opensource.org/licenses/MIT MIT
  * @version   Release:0.1.2
  * @link      https://github.com/jcupitt/php-vips
+ *
+ * @property int $bands
+ * @property int $width
+ * @property int $height
+ * @property string $interpretation
+ *
+ * @method Image cast($format, array $options = [])
+ * @method Image embed(int $x, int $y, int $width, int $height, array $options = [])
+ * @method Image extract_band(int $band, array $options = [])
+ * @method array min(array $options)
+ * @method array max(array $options)
+ * @method Image round(string $round, array $options = [])
+ * @method Image bandbool(string $boolean, array $options = [])
+ * @method Image complexget(string $get, array $options = [])
+ * @method Image complex(string $cmplx, array $options = [])
+ * @method Image math(string $math, array $options = [])
+ * @method Image morph(Image $mask, string $morph)
+ * @method Image rank(int $width, int $height, int $index)
+ * @method Image flip($direction, array $options = [])
+ * @method Image rot($angle, array $options = [])
+ * @method static Image black(int $width, int $height, array $options = [])
+ * @method static Image linear(float $a, float $b, array $options = [])
```

Comment:
I'm planning to generate most of these automatically from introspection, I wouldn't spend too much time writing them by hand. For example, the Ruby binding has this file:

https://github.com/jcupitt/ruby-vips/blob/master/lib/vips/methods.rb

Generated automatically by this code:

https://github.com/jcupitt/ruby-vips/blob/master/lib/vips/image.rb#L1282

It should be easy to adapt to PHP.